ophyd/status.py

```diff
@@ -55,35 +60,85 @@ def _timeout_thread(self):
             self._timeout_thread = None
 
     @_locked
-    def _finished(self, success=True, **kwargs):
+    def _finished(self, *, success=True, timestamp=None, **kwargs):
```

Comment:
I think not locking during the callbacks could help avoid some deadlock conditions:

```python
def _finished(self, *, success=True, timestamp=None, **kwargs):
    # args/kwargs are not really used, but are passed - because pyepics
    # gives in a bunch of kwargs that we don't care about
    with self._lock:
        if self.done:
            return

        self.success = success
        self.done = True
        if timestamp is None:
            timestamp = time.time()
        self.finish_ts = timestamp

    for cb in list(self._callbacks):
        cb()
        self._callbacks.remove(cb)
```